### ISSUE TITLE
Preliminary Suggestions on Model Configuration

### ISSUE THREAD

**Issue by yu45020** [nmhkahn/CARN-pytorch#8]
Thanks for sharing your implementation on the model. The paper seems to suggest the model is used on mobile devices. But I assume it is used in standard desktops/laptops and add more features. I have very limited GPU, so the following experimental results are preliminary and may well be wrong. 

1. Append [Concurrent Spatial and Channel Squeeze & Excitation](https://arxiv.org/abs/1803.02579) after each residual blocks; 

2. upscale the input image and add it to the final convolution output; 

3. replace all 1x1 convolution by 3x3 with groups. 4 groups have shown clear improvement. 

![img](https://raw.githubusercontent.com/yu45020/Waifu2x/master/Readme_imgs/CARN/carn_plot_loss_files/carn_plot_loss_6_0.png)

![img](https://raw.githubusercontent.com/yu45020/Waifu2x/master/Readme_imgs/CARN/carn_plot_loss_files/carn_plot_loss_32_0.png)

The standard blue line is your model configuration. "se" means "Concurrent Spatial and Channel Squeeze & Excitation", and "res" means residual add on up-scaled input image. "group1" means all 1x1 conv are replaced by 3x3.

My GPU doesn't have enough memory for validation test while training, so I only check the L1 loss.  You may find more details [here](https://github.com/yu45020/Waifu2x/blob/master/Readme_imgs/CARN/carn_plot_loss.md#major-comparison). 
 
Hope they are useful. 

Please tell me if I am wrong. Thanks.

**Comment by nmhkahn** [comment]
Hi. Thanks for the suggestion.

We don't have any experience adopting the SENet architecture, but if this performs better without extra burden, it gonna be a good idea.
Likewise, replacing the 1x1 to grouped 3x3 can be a one capable option in terms of the performance-efficiency trade-off.
For the second suggestion, we did a similar approach to our extend paper and perform better (but not a dramatic)
And unfortunately, because I have another project I'm not sure can handle above stuffs, but thanks anyway 👍

**Comment by yu45020** [comment]
Thanks for sharing your experience. I don't mean you try those proposals. Instead, I find it very help by sharing what architecture works and what doesn't. The SENet is suggested in an issue comment from [nagadomi's waifu2x](https://github.com/nagadomi/waifu2x/issues/231#issuecomment-440842370). 

I am curious on atrous convolution. Image segment and classification get obvious improvement by simply replacing strikes with atrous convolution. But it seems has little improvement in super resolution, and sometime it makes the model worse. The holes in convolution may blur images. If you have any experience about it during experiments, I would like to hear your ideas.

**Comment by nmhkahn** [comment]
We conducted a toy example using atrous convolution and find that it doesn't provide concrete improvement for the SR task. Not sure why, but I guess that when training the SR model, we usually crop the image to small patches. So the benefit of using atrous convolution might be discarded since the additional gain of receptive field doesn't help anymore.

**Comment by yu45020** [comment]
I get the same result that atrous conv doesn't have obvious improvement and [sometime it gets worse](https://github.com/yu45020/Waifu2x/blob/master/Readme_imgs/CARN/carn_plot_loss.md#atrous-convolution). In my experiment setup, I use 128x128 image as input and various atrous rates in the global block level. I even try 5 blocks with rates 1-2-4-2-1, but its improvement might well come from extra parameters because the model size is almost doubled. 128x128 shall be large enough given the small rates. 

In the local block (residual part), I also try 1 atrous conv followed by standard convolution, but it doesn't help.  If holes in the atrous conv expands field of view, then the followed standard conv shall cover details between the holes. The experiments suggest I am wrong. 

![img](https://raw.githubusercontent.com/yu45020/Waifu2x/master/Readme_imgs/CARN/carn_plot_loss_files/carn_plot_loss_20_0.png)
